packages/protocol/contracts/common/GasPriceMinimum.sol

```diff
@@ -153,14 +154,20 @@ contract GasPriceMinimum is Ownable, Initializable, UsingRegistry {
     view
     returns (uint256)
   {
-    FractionUtil.Fraction memory one = FractionUtil.Fraction(1, 1);
-    FractionUtil.Fraction memory blockDensity = FractionUtil.Fraction(
-      blockGasTotal,
-      blockGasLimit
-    );
-    FractionUtil.Fraction memory adjustment = one.add(adjustmentSpeed.mul(blockDensity)).sub(
-      adjustmentSpeed.mul(targetDensity)
+    FixidityLib.Fraction memory blockDensity = FixidityLib.newFixedFraction(
+      blockGasTotal, blockGasLimit
     );
-    return adjustment.mul(gasPriceMinimum).add(1);
+    bool densityGreaterThanTarget = blockDensity.gt(targetDensity);
+    FixidityLib.Fraction memory densityDelta = densityGreaterThanTarget ?
+      blockDensity.subtract(targetDensity) :
+      targetDensity.subtract(blockDensity);
+    FixidityLib.Fraction memory adjustment = densityGreaterThanTarget ?
+      FixidityLib.fixed1().add(adjustmentSpeed.multiply(densityDelta)) :
+      FixidityLib.fixed1().subtract(adjustmentSpeed.multiply(densityDelta));
+
+    return adjustment
+      .multiply(FixidityLib.newFixed(gasPriceMinimum))
```

Comment:
syntactic sugar: maybe we have Fixidity overload arithmetic operation and support uint. we can even improve computations for those cases